master/internal/api_agents.go

```diff
@@ -10,7 +10,15 @@ import (
 func (a *apiServer) GetAgents(
 	_ context.Context, req *apiv1.GetAgentsRequest) (resp *apiv1.GetAgentsResponse, err error) {
 	err = a.actorRequest("/agents", req, &resp)
-	return resp, err
+	if err != nil {
+		return nil, err
```

Comment:
question: Do we want to provide a status code or any other information here?

Reply:
actorRequest will wrap the error for us already.